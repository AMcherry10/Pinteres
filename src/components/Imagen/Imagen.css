.Imagen-img{
  width: 100%;
  height: auto;
}

.Imagen-content{
  display: inline-flex;
  flex-direction: column;
  width: 100%;
  padding: 10px;
  margin: 5px 0;
  background-color: lightgray;
  border-radius: 5px;
}

