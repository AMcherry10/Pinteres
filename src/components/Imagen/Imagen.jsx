import React, { Component } from "react";
import { Button, Modal, ModalHeader, ModalBody, ModalFooter } from "reactstrap";
import "./Imagen.css";

class Imagen extends Component {
  constructor(props) {
    super(props);
    this.state = {
      modal: false
    };

    this.toggle = this.toggle.bind(this);
  }

  toggle() {
    this.setState({
      modal: !this.state.modal
    });
  }

  render() {
    return (
      <div className="Imagen-content">
        <img src={this.props.img} alt="foto" className="Imagen-img" onClick={this.toggle} />
        <h5>{this.props.titulo}</h5>
        <h6>{this.props.likes}</h6>
        <p>{this.props.user}</p>

        <Modal isOpen={this.state.modal} toggle={this.toggle}>
          <ModalHeader toggle={this.toggle}>
            <h5>{this.props.titulo}</h5>
          </ModalHeader>
          <ModalBody>
            <img src={this.props.img} alt="foto" className="Imagen-img" />
          </ModalBody>
          <ModalFooter className="text-left">
            <p className="userModal">Usuario: {this.props.user} <br/> Likes: {this.props.likes}</p>            
          </ModalFooter>
        </Modal>
      </div>
    )
  }
}

export default Imagen;