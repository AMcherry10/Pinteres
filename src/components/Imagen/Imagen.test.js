import React from "react";
import Enzyme, { shallow, mount, render } from "enzyme";
import renderer from "react-test-renderer";
import Imagen from "../Imagen/Imagen";
import Adapter from "enzyme-adapter-react-16"

Enzyme.configure({ adapter: new Adapter() });

describe("Imagen Component", () => {
 
  // make our assertion and what we expect to happen 
  it("hace render de componente", () => {
    expect(shallow(<Imagen />).find("div.Imagen-content").exists()).toBe(true)
  })

  it("render correcto", () => {
    const component = renderer.create(
      <Imagen />,
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  })
 })