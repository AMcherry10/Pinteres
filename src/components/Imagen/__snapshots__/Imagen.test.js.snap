// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Imagen Component render correcto 1`] = `
<div
  className="Imagen-content"
>
  <img
    alt="foto"
    className="Imagen-img"
    onClick={[Function]}
    src={undefined}
  />
  <h5 />
  <h6 />
  <p />
</div>
`;
